File: node/zkgroup/profiles/ProfileKeyCommitment.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';

export default class ProfileKeyCommitment extends ByteArray {
  static SIZE = 97;

  constructor(contents: Buffer) {
    super(contents, ProfileKeyCommitment.SIZE, true);
    NativeImpl.ProfileKeyCommitment_CheckValidContents(contents);
  }
}

File: java/shared/java/org/signal/libsignal/zkgroup/profiles/ProfileKeyCommitment.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.zkgroup.profiles;

import static org.signal.libsignal.internal.FilterExceptions.filterExceptions;

import org.signal.libsignal.internal.Native;
import org.signal.libsignal.zkgroup.InvalidInputException;
import org.signal.libsignal.zkgroup.internal.ByteArray;

public final class ProfileKeyCommitment extends ByteArray {
  public ProfileKeyCommitment(byte[] contents) throws InvalidInputException {
    super(contents);
    filterExceptions(
        InvalidInputException.class,
        () ->
            filterExceptions(
                InvalidInputException.class,
                () -> Native.ProfileKeyCommitment_CheckValidContents(contents)));
  }
}
